lib/executor/variable_arrival_rate_test.go

```diff
@@ -236,36 +218,17 @@ func testVariableArrivalRateSetup(t *testing.T, vuFn func(context.Context, chan<
 		PreAllocatedVUs: null.IntFrom(10),
 		MaxVUs:          null.IntFrom(20),
 	}
-	logHook := &testutils.SimpleLogrusHook{HookedLevels: []logrus.Level{logrus.WarnLevel}}
-	testLog := logrus.New()
-	testLog.AddHook(logHook)
-	testLog.SetOutput(ioutil.Discard)
-	logEntry := logrus.NewEntry(testLog)
-	es := lib.NewExecutionState(lib.Options{}, 10, 50)
-	runner := lib.MiniRunner{
-		Fn: vuFn,
-	}
-
-	es.SetInitVUFunc(func(_ context.Context, _ *logrus.Entry) (lib.VU, error) {
-		return &lib.MiniRunnerVU{R: runner}, nil
-	})
-
-	initializeVUs(ctx, t, logEntry, es, 10)
-
-	executor, err := config.NewExecutor(es, logEntry)
-	require.NoError(t, err)
-	err = executor.Init(ctx)
-	require.NoError(t, err)
-	return ctx, cancel, executor, logHook
 }
 
 func TestVariableArrivalRateRunNotEnoughAllocatedVUsWarn(t *testing.T) {
 	t.Parallel()
-	var ctx, cancel, executor, logHook = testVariableArrivalRateSetup(
-		t, func(ctx context.Context, out chan<- stats.SampleContainer) error {
+	var ctx, cancel, executor, logHook = setupExecutor(
+		t, getTestVariableArrivalRateConfig(),
+		func(ctx context.Context, out chan<- stats.SampleContainer) error {
 			time.Sleep(time.Second)
```

Comment:
Could you please explain why we need sleeping here?

Reply:
I would ask the original author that question (@MStoykov) ;), but I've used a similar approach in other tests to avoid hogging the CPU needlessly and be able to produce more testable / deterministic results.

Reply:
If you go down the test you will see that we are testing what happens if VUs iterations aren't fast enough and we run out of VUs. I could've made it do something .. like execute JS parsing 5mb CSV  .. but that will just waste resources ... all I did care is that it takes long enough, so sleep seems like a good idea

Reply:
@MStoykov Thanks 👍 
@imiric Could you please add some comments for that sleep, because it's not very clear for the reader when looking at the code at first.

Reply:
@cuonglm Do we really need a comment for this? This is the default function executed by the VU. It's understandable it needs to do some work, and a fixed sleep is the most deterministic and unobtrusive way of doing that.

If we add a comment here, I suppose it should be in all other tests as well, since this pattern is frequently used? If so, please let me know what would explain it for all cases. "fake work for VU"?

Reply:
Then why not make the function global, then you only need to comment in one place.

Reply:
Because the work function depends on the test and will not be the same for all. Parametrizing it for purposes of not repeating a comment seems overkill.

Reply:
Agree